File: 0701_0800/0791/0791.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    s = []
    if n > 8:
        s.append(n - 8)
    if n % 8 != 1:
        s.append(n - 1)
    if n % 8:
        s.append(n + 1)
    if n < 57:
        s.append(n + 8)
    q.write(' '.join(str(x) for x in s))
    

File: 0701_0800/0791/0791.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n;

	fscanf(f, "%d", &n);
	if (n > 8)
		fprintf(q, "%d ", n - 8);
	if (n % 8 != 1)
		fprintf(q, "%d ", n - 1);
	if (n % 8)
		fprintf(q, "%d ", n + 1);
	if (n < 57)
		fprintf(q, "%d ", n + 8);
	return 0;
}
